engine/api/src/main/java/org/apache/cloudstack/engine/subsystem/api/storage/DataStoreCapabilities.java

```diff
@@ -40,5 +40,14 @@ public enum DataStoreCapabilities {
     /**
      * indicates that this driver supports reverting a volume to a snapshot state
      */
-    CAN_REVERT_VOLUME_TO_SNAPSHOT
+    CAN_REVERT_VOLUME_TO_SNAPSHOT,
+    /**
+    * indicates that the driver supports copying snapshot between zones on pools of the same type
+    */
+    CAN_COPY_SNAPSHOT_BETWEEN_ZONES_AND_SAME_POOL_TYPE,
+    /**
+     * indicates that the storage does not need to delete the snapshot when creating a volume/template from it
+     * and the setting `snapshot.backup.to.secondary` is enabled
+     */
+    KEEP_SNAPSHOT_ON_PRIMARY_AND_BACKUP
```

Comment:
this seems to be not a datastore capability (maybe, some flag needed in snapshot to keep in primary and backup storage, check if that can be specified through API cmd)

Reply:
@sureshanaparti, I don't want to give this option to the users, that's how the StorPool plugin works. Would it be better to check that the pool type is StorPool instead of adding a new capability?

Reply:
maybe, that's better. in case StorPool wants to keep as primary & backup.

Reply:
thanks @sureshanaparti, I'll change it

Reply:
changed in the latest commits